src/analysis/record.rs

```diff
@@ -176,20 +176,16 @@ pub fn new(env: &Env, obj: &GObject) -> Option<Info> {
             || !specials.has_trait(special_functions::Type::Free))
     {
         if let Some((_, get_type_version)) = glib_get_type {
-            if get_type_version > version {
-                // FIXME: Ideally we would update it here but that requires fixing *all* the
-                // versions of functions in this and other types that use this type somewhere in
-                // the signature. Similar code exists for before the analysis already but that
-                // doesn't apply directly here.
-                //
-                // As the get_type function only has a version if explicitly configured let's just
-                // panic here. It's easy enough for the user to move the version configuration from
-                // the function to the type.
-                panic!(
-                    "Have to use get_type function for {} but version is higher than for the type ({:?} > {:?})",
-                    full_name, get_type_version, version
-                );
-            }
+            // FIXME: Ideally we would update it here but that requires fixing *all* the
+            // versions of functions in this and other types that use this type somewhere in
+            // the signature. Similar code exists for before the analysis already but that
+            // doesn't apply directly here.
+            //
+            // As the get_type function only has a version if explicitly configured let's just
+            // panic here. It's easy enough for the user to move the version configuration from
+            // the function to the type.
+            assert!(get_type_version <= version, "Have to use get_type function for {} but version is higher than for the type ({:?} > {:?})",
```

Comment:
Can you keep the argument formatting (ie one per line) like we had before please? It's a bit hard to read currently.

Reply:
I made some changes to it. I hope I understood you correctly and it was you requested. I am so used to the auto-formatting, that I never check the formatting.

Reply:
Unfortunately `rustfmt` doesn't seem to be able to deal with long string literals in macro invocations and just ignores the entire call and block(s!) around it.

Haven't checked if there's an upstream issue/report for this, though. Alas, the wrapped format like:

```rust
assert!("some very long thing",
        aligned_argument_on_the_next_line); // closing parenthesis on the same line
```

is more C/C++ like :)

Reply:
Just to make sure, are you fine with the current changes?

Reply:
Yeah, this looks like what `rustfmt` would make out of it. When unsure, temporarily shorten the string literal to make `rustfmt` do its thing. In this case it seems to turn it into:

```rust
            assert!(
                get_type_version <= version,
                "Have to use get_type function for {} but version is higher than for the type ({:?} > {:?})",
                full_name,
                get_type_version,
                version(,)
            );
```

(It doesn't insert a comma as that sometimes messes with macro matching, even though it's more consistent with function calls. I'd insert it here by hand :slightly_smiling_face:)